lib/ansible/modules/cloud/azure/GUIDELINES.md

```diff
@@ -0,0 +1,154 @@
+```
+
+## Documenting Return Values
+
+Facts module should include full documentation of return value, as in following example:
+
+```
+RETURN = '''
+servers:
+    description: A list of dictionaries containing facts for PostgreSQL servers.
+    returned: always
+    type: complex
+    contains:
+        id:
+            description:
+                - Resource ID
+            returned: always
+            type: str
+            sample: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx/resourceGroups/TestGroup/providers/Microsoft.DBforPostgreSQL/servers/postgreabdud1223
+        resource_group:
+            description:
+                - Resource group name.
+            returned: always
+            type: str
+            sample: TestGroup
+        name:
+            description:
+                - Resource name.
+            returned: always
+            type: str
+            sample: postgreabdud1223
+        location:
+            description:
+                - The location the resource resides in.
+            returned: always
+            type: str
+            sample: eastus
+        sku:
+            description:
+                - The SKU of the server.
+            returned: always
+            type: complex
+            contains:
+                name:
+                    description:
+                        - The name of the SKU
+                    returned: always
+                    type: str
+                    sample: GP_Gen4_2
+                tier:
+                    description:
+                        - The tier of the particular SKU
+                    returned: always
+                    type: str
+                    sample: GeneralPurpose
+                capacity:
+                    description:
+                        - The scale capacity.
+                    returned: always
+                    type: int
+                    sample: 2
+        version:
+            description:
+                - Server version.
+            returned: always
+            type: str
+            sample: "9.6"
+        user_visible_state:
+            description:
+                - A state of a server that is visible to user.
+            returned: always
+            type: str
+            sample: Ready
+        fully_qualified_domain_name:
+            description:
+                - The fully qualified domain name of a server.
+            returned: always
+            type: str
+            sample: postgreabdud1223.postgres.database.azure.com
+'''
+```
+
+## Samples
+
+Samples should include all possible combinations of input parameters showing how to:
+- query all the resources of given type in an entire subscription
+- query all the resources of given type in the resource group
+- query specific resource
+- filter resources by tags
+
+## Testing
+
+Following rules should apply to tests:
+- Do not create a separate test, add **azure_xx_facts** tests to **azure_xx** module tests
```

Comment:
Do not create a separate test, add **`azure_rm_xx_facts`** tests to **`/test/integration/targets/azure_rm_xx`** module tests